src/pages/Home/components/Portfolio/Prompt/Prompt.tsx

```diff
@@ -0,0 +1,459 @@
+      .replace(
+        '__NETWORKS___',
+        JSON.stringify(
+          networks.map((n) => ({
+            id: n.caipId,
+            name: n.chainName,
+            isTestnet: n.isTestnet,
+          })),
+        ),
+      )
+      .replace(
+        '__CURRENT_NETWORK___',
+        JSON.stringify({
+          id: network.caipId,
+          name: network.chainName,
+          isTestnet: network.isTestnet,
+        }),
+      )
+      .replace('__CONTACTS__', JSON.stringify(contacts))
+      .replace(
+        '__ACCOUNTS__',
+        JSON.stringify(
+          [
+            ...Object.values(accounts.primary).flat(),
+            ...Object.values(accounts.imported),
+          ].map((a) => ({ name: a.name, id: a.id, address: a.addressC })),
+        ),
+      );
+  }, [tokens, network, contacts, accounts, networks]);
+
+  const model = useMemo(() => {
+    console.log('SYSTEM PROMPT', systemPrompt);
+    return genAI.getGenerativeModel({
+      model: 'gemini-2.0-flash',
+      generationConfig: {
+        temperature: 0.5,
+        topP: 0.95,
+        topK: 40,
+        maxOutputTokens: 8192,
+      },
+      tools: [
+        {
+          functionDeclarations,
+        },
+      ],
+      toolConfig: {
+        functionCallingConfig: {
+          mode: FunctionCallingMode.AUTO,
+        },
+      },
+      systemInstruction: systemPrompt,
+    });
+  }, [systemPrompt]);
+
+  const prompt = useCallback(
+    async (message: string) => {
+      setPrompts((prev) => {
+        return [...prev, { type: 'user', content: message }];
+      });
+      setInput('');
+
+      const chat = model.startChat();
+
+      // Send the message to the model.
+      const result = await chat.sendMessage(message);
+
+      // For simplicity, this uses the first function call found.
+      const call = result?.response?.functionCalls()?.[0];
+
+      if (call) {
+        // Call the executable function named in the function call
+        // with the arguments specified in the function call and
+        // let it call the hypothetical API.
+        try {
+          const apiResponse = await functions[call.name](call.args);
+
+          console.log('PROMPT apiResponse', apiResponse);
+          // Send the API response back to the model so it can generate
+          // a text response that can be displayed to the user.
+          const result2 = await chat.sendMessage([
+            {
+              functionResponse: {
+                name: 'send',
+                response: apiResponse,
+              },
+            },
+          ]);
+          console.log('PROMPT result2', result2);
+
+          // Log the text response.
+          setPrompts((prev) => {
+            return [
+              ...prev,
+              { type: 'system', content: result2.response.text() },
+            ];
+          });
+        } catch (e: any) {
+          const errorMessage =
+            'code' in e
+              ? errorValues[e.code]?.message || 'Unkown error happened'
```

Comment:
```suggestion
              ? errorValues[e.code]?.message || 'Unknown error happened'
```

Should we translate it, too?